inet:ipv6 tests fail on osx

```
$ py.test synapse/tests/test_types.py 
========================================================================================== test session starts ===========================================================================================
platform darwin -- Python 3.5.1, pytest-3.0.2, py-1.4.31, pluggy-0.3.1
rootdir: /Users/adam.heinz/Documents/GitHub/synapse, inifile: tox.ini
collected 22 items 

synapse/tests/test_types.py .........F............

================================================================================================ FAILURES ================================================================================================
_________________________________________________________________________________ DataTypesTest.test_datatype_inet_ipv6 __________________________________________________________________________________

self = <synapse.tests.test_types.DataTypesTest testMethod=test_datatype_inet_ipv6>

    def test_datatype_inet_ipv6(self):
        tlib = s_types.TypeLib()

        self.assertRaises(BadTypeValu, tlib.getTypeParse, 'inet:ipv6', 'newp' )
        self.assertRaises(BadTypeValu, tlib.getTypeNorm, 'inet:srv6', 'newp' )

        self.eq( tlib.getTypeParse('inet:ipv6', 'AF:00::02'), 'af::2')
        self.eq( tlib.getTypeNorm('inet:ipv6', 'AF:00::02'), 'af::2')
        self.eq( tlib.getTypeRepr('inet:ipv6', 'af::2'), 'af::2')

>       self.eq( tlib.getTypeNorm('inet:ipv6', '2001:db8::1:1:1:1:1'), '2001:db8:0:1:1:1:1:1')

synapse/tests/test_types.py:132: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
synapse/tests/common.py:65: in eq
    self.assertEqual(x,y)
E   AssertionError: '2001:db8::1:1:1:1:1' != '2001:db8:0:1:1:1:1:1'
E   - 2001:db8::1:1:1:1:1
E   + 2001:db8:0:1:1:1:1:1
E   ?          +
========================================================================================= pytest-warning summary ========================================================================================
```

--- a/synapse/models/inet.py
+++ b/synapse/models/inet.py
@@ -2,6 +2,7 @@ import socket
 import struct
 import hashlib
 import logging
+import ipaddress
 
 import regex
 
@@ -111,8 +112,17 @@ def ipv6norm(text):
         text = ipv6norm(text)
 
     '''
-    # use inet_ntop / inet_pton from synapse.lib.socket for portability
-    return s_socket.inet_ntop(socket.AF_INET6, s_socket.inet_pton(socket.AF_INET6, text))
+    # This used to use socket.inet_pton for normalizing
+    # ipv6 addresses. There's a bug in macOS where it
+    # doesn't abbreviate the zeroes properly. See RFC
+    # 5952 for IPv6 address text representation
+
+    v6addr = ipaddress.IPv6Address(text)
+    v4addr = v6addr.ipv4_mapped
+    if v4addr is not None:
+        return '::ffff:%s' % (v4addr)
+
+    return v6addr.compressed
 
 class IPv6Type(DataType):
     def repr(self, valu):

--- a/synapse/tests/test_model_inet.py
+++ b/synapse/tests/test_model_inet.py
@@ -96,7 +96,37 @@ class InetModelTest(SynTest):
         self.eq(tlib.getTypeRepr('inet:ipv6', 'af::2'), 'af::2')
 
         self.eq(tlib.getTypeNorm('inet:ipv6', '2001:db8::1:1:1:1:1')[0], '2001:db8:0:1:1:1:1:1')
+
+        # Specific examples given in RFC5952
+        # Section 1
+        self.eq(tlib.getTypeNorm('inet:ipv6', '2001:db8:0:0:1:0:0:1')[0], '2001:db8::1:0:0:1')
+        self.eq(tlib.getTypeNorm('inet:ipv6', '2001:0db8:0:0:1:0:0:1')[0], '2001:db8::1:0:0:1')
+        self.eq(tlib.getTypeNorm('inet:ipv6', '2001:db8::1:0:0:1')[0], '2001:db8::1:0:0:1')
+        self.eq(tlib.getTypeNorm('inet:ipv6', '2001:db8::0:1:0:0:1')[0], '2001:db8::1:0:0:1')
+        self.eq(tlib.getTypeNorm('inet:ipv6', '2001:0db8::1:0:0:1')[0], '2001:db8::1:0:0:1')
+        self.eq(tlib.getTypeNorm('inet:ipv6', '2001:db8:0:0:1::1')[0], '2001:db8::1:0:0:1')
+        self.eq(tlib.getTypeNorm('inet:ipv6', '2001:DB8:0:0:1::1')[0], '2001:db8::1:0:0:1')
+
+        # Section 2.1
+        self.eq(tlib.getTypeNorm('inet:ipv6', '2001:DB8:0:0:1:0000:0000:1')[0], '2001:db8::1:0:0:1')
+
+        # Section 2.2
+        self.raises(BadTypeValu, tlib.getTypeNorm, 'inet:ipv6', '::1::')
+
+        # Section 4.1
+        self.eq(tlib.getTypeNorm('inet:ipv6', '2001:0db8::0001')[0], '2001:db8::1')
+
+        # Section 4.2.1
+        self.eq(tlib.getTypeNorm('inet:ipv6', '2001:db8:0:0:0:0:2:1')[0], '2001:db8::2:1')
+
+        # Section 4.2.2
         self.eq(tlib.getTypeNorm('inet:ipv6', '2001:db8:0:1:1:1:1:1')[0], '2001:db8:0:1:1:1:1:1')
+
+        # Section 4.2.3
+        self.eq(tlib.getTypeNorm('inet:ipv6', '2001:0:0:1:0:0:0:1')[0], '2001:0:0:1::1')
+        self.eq(tlib.getTypeNorm('inet:ipv6', '2001:db8:0:0:1:0:0:1')[0], '2001:db8::1:0:0:1')
+
+        self.eq(tlib.getTypeNorm('inet:ipv6', '::ffff:1.2.3.4')[0], '::ffff:1.2.3.4')
 
         self.eq(tlib.getTypeNorm('inet:ipv6', '2001:db8::0:1')[0], '2001:db8::1')
         self.eq(tlib.getTypeNorm('inet:ipv6', '2001:db8:0:0:0:0:2:1')[0], '2001:db8::2:1')

